src/CoordinateSystemsCore/CoordinateSystemsCore.jl

```diff
@@ -0,0 +1,33 @@
+module CoordinateSystemsCore
+
+using StaticArrays
+
+export AbstractCoordinateSystem, Cartesian, Spherical
+export point, coordinate, coordinate_system, with_coordinate_system
+export AbstractPoint, Point
+
+# We accept several user input types, but always store and compute as SVector
+const CoordinateType{N} = Union{NTuple{N}, SVector{N}}
+
+abstract type AbstractCoordinateSystem{N} end
+abstract type FixedOriginCoordinateSystem{N} <: AbstractCoordinateSystem{N} end
```

Comment:
Can you give an example of a reference system without fixed origin?

Reply:
Points in the unit sphere. It must be defined as a manifold with some atlas (coordinate charts).